lints/lint_dnsname_bad_character_in_label.go

```diff
@@ -1,50 +1,36 @@
 package lints
 
 import (
+	"regexp"
+
 	"github.com/zmap/zcrypto/x509"
 	"github.com/zmap/zlint/util"
-	"regexp"
-	"strings"
 )
 
 type DNSNameProperCharacters struct {
 	CompiledExpression *regexp.Regexp
 }
 
 func (l *DNSNameProperCharacters) Initialize() error {
-	const dnsLabelRegex = "^[A-Za-z0-9*_-]+$"
-	l.CompiledExpression = regexp.MustCompile(dnsLabelRegex)
-	return nil
+	const dnsNameRegexp = `^(\*\.)?(\?\.)?(A-Za-z0-9*_-]+\.)*[A-Za-z0-9*_-]*$`
+	var err error
+	l.CompiledExpression, err = regexp.Compile(dnsNameRegexp)
```

Comment:
This regex can be moved to a global variable assigned to the result of `MustCompile`, which has the side effect of ensuring a panic when the program starts if there is a typo in the pattern.

Reply:
In my book, `Initialize()` returns an error because can fail like in this case. What is the point of returning an error from the function if prefered mode of handling an error is panicking?

It is your repo, and I will do what guys ask me to do, but I don't think that panicking is a better way to handle errors than returning an error. 

Additionally the init func calls `RegisterLint()`, which in turn calls that lint's `Initialize()` and panics on error returned from `Initialize()`

So yeah, the code panics anyway if regex could not be compiled.